File: SDKs/vmware.local/vrni/1.0.0/typescript/src/sdk/models/operations/enablebrocadeswitch.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class EnableBrocadeSwitchPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class EnableBrocadeSwitchSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth: shared.SchemeApiKeyAuth;
}


export class EnableBrocadeSwitchRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: EnableBrocadeSwitchPathParams;

  @SpeakeasyMetadata()
  security: EnableBrocadeSwitchSecurity;
}


export class EnableBrocadeSwitchResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/vmware.local/vrni/1.0.0/python/src/sdk/models/operations/enablebrocadeswitch.py
from dataclasses import dataclass, field
from sdk.models import shared


@dataclass
class EnableBrocadeSwitchPathParams:
    id: str = field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass
class EnableBrocadeSwitchSecurity:
    api_key_auth: shared.SchemeAPIKeyAuth = field(metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclass
class EnableBrocadeSwitchRequest:
    path_params: EnableBrocadeSwitchPathParams = field()
    security: EnableBrocadeSwitchSecurity = field()
    

@dataclass
class EnableBrocadeSwitchResponse:
    content_type: str = field()
    status_code: int = field()
    
